rider/src/main/kotlin/com/jetbrains/rider/settings/RiderUnitySettings.kt

```diff
@@ -3,16 +3,16 @@ package com.jetbrains.rider.settings
 import com.intellij.ide.util.PropertiesComponent
 import com.intellij.openapi.project.Project
 import com.jetbrains.rd.util.lifetime.Lifetime
-import com.jetbrains.rdclient.util.idea.LifetimedProjectComponent
 import com.jetbrains.rd.util.reactive.Signal
+import com.jetbrains.rdclient.util.idea.ProtocolSubscribedProjectComponent
 
-class RiderUnitySettings(project: Project) : LifetimedProjectComponent(project) {
+class RiderUnitySettings(project: Project) : ProtocolSubscribedProjectComponent(project) {
```

Comment:
it doesn't subscribe to the protocol, has no init {}, stores no state apart from `properties = PropertiesComponent.getInstance()` which is also bad. 

There's no need for this class, it has no payload BooleanViewProperty does.

`BooleanViewProperty` can be renamed to something matching its meaning like `PropertiesBackedViewableBooleanSetting`. But I'd not recommend using this as a pattern. As it doesn't react to changes in PropertiesComponent or external modifications in xml PropertiesComponent uses for storage.